clash-lib/src/Clash/Netlist/Types.hs

```diff
@@ -135,6 +135,8 @@ data HWType
   -- that vector.
   | String
   -- ^ String type
+  | Integer
+  -- ^ Integer type
```

Comment:
Why do we need `Integer` as a hwtype?

Reply:
I guess the documentation should be extended that this `Integer` is for internal use only, and that it is not, nor should ever be used as the type representing the Haskell `Integer` type.